src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ObjectWriter/Dwarf/DwarfFde.cs

```diff
@@ -0,0 +1,113 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Buffers;
+using static ILCompiler.ObjectWriter.DwarfNative;
+
+namespace ILCompiler.ObjectWriter
+{
+    internal sealed class DwarfFde
+    {
+        public readonly DwarfCie Cie;
+        public string PcStartSymbolName;
+        public long PcStartSymbolOffset;
+        public ulong PcLength;
+        public string LsdaSymbolName;
+        public string PersonalitySymbolName;
+        public readonly byte[] Instructions;
+
+        public DwarfFde(DwarfCie cie, byte[] instructions)
+        {
```

Comment:
There is only one place where these are constructed. I think it could be better to pass all the parts to the constructor and make fields readonly.